src/instruments/src/SD/Pages/Fctl/Fctl.tsx

```diff
@@ -0,0 +1,767 @@
+    // const [bluePumpLowPressure] = useSimVar('L:A32NX_HYD_BLUE_EPUMP_LOW_PRESS', 'boolean', 500);
+
+    // function checkPumpLowPressure(pump) {
+    //     switch (pump) {
+    //     case 'GREEN':
+    //         return (greenPumpLowPressure && greenPumpActive) || !greenPumpActive;
+    //     case 'BLUE':
+    //         return (bluePumpLowPressure && bluePumpActive) || !bluePumpActive;
+    //     case 'YELLOW':
+    //         return (yellowPumpLowPressure && yellowPumpActive) || !yellowPumpActive;
+    //     default:
+    //         return 1;
+    //     }
+    // }
+
+    // const [hydraulicGAvailable] = checkPumpLowPressure('GREEN');
+    // const [hydraulicYAvailable] = checkPumpLowPressure('YELLOW');
+    // const [hydraulicBAvailable] = checkPumpLowPressure('BLUE');
+
+    const [engine1State] = useSimVar('TURB ENG N2:1', 'Percent', 1000);
+    const [engine2State] = useSimVar('TURB ENG N2:2', 'Percent', 1000);
+
+    const hydraulicGAvailable = engine1State > 15;
+    const hydraulicYAvailable = engine2State > 15;
+    const hydraulicBAvailable = engine1State > 15 || engine2State > 15;
+
+    const [leftSpoilerState] = useSimVar('SPOILERS LEFT POSITION', 'percent over 100', 50);
+    const [rightSpoilerState] = useSimVar('SPOILERS RIGHT POSITION', 'percent over 100', 50);
+    const [spoilerHandleState] = useSimVar('SPOILERS HANDLE POSITION', 'percent over 100', 100);
+
+    const [groundSpeedState] = useSimVar('SURFACE RELATIVE GROUND SPEED', 'feet_per_second', 50);
+
+    const [spoilersArmedState] = useSimVar('SPOILERS ARMED', 'boolean', 500);
+
+    const [onGroundState] = useSimVar('SIM ON GROUND', 'boolean', 500);
+
+    const [engine1ModeState] = useSimVar('GENERAL ENG THROTTLE MANAGED MODE:1', 'number', 1000);
+    const [engine2ModeState] = useSimVar('GENERAL ENG THROTTLE MANAGED MODE:2', 'number', 1000);
+
+    const spoilersArmed = !!(spoilersArmedState
+        && onGroundState
+        && groundSpeedState > 25
+        && engine1ModeState <= 2
+        && engine2ModeState <= 2);
+
+    const speedBrakeY = 104;
+    const leftSpeedBrakeX = 103;
+    const rightSpeedBrakeX = 497;
+    const speedBrakeXChange = 38;
+    const spoilers: Array<any> = [];
+    for (let i = 0; i < 5; i += 1) {
+        const YCoord = speedBrakeY - (i * 5);
+        const YCoordW = YCoord - 9;
+        const leftX = leftSpeedBrakeX + (i * speedBrakeXChange);
+        const rightX = rightSpeedBrakeX - (i * speedBrakeXChange);
+        const index = 5 - i;
+        spoilers.push(<Spoiler
+            leftorright="left"
+            index={index}
+            x={leftX}
+            y={YCoord}
+            yw={YCoordW}
+            hydAvail={[hydraulicGAvailable, hydraulicBAvailable, hydraulicYAvailable]}
+            speedbrake={spoilerHandleState}
+            spoilerpos={leftSpoilerState}
+            ailpos={aileronLeftDeflectionState}
+            spoilersArmed={spoilersArmed}
```

Comment:
To suppress some potential React warnings, pass `key={index}` here (whenever using JSX in an array).